test/extensions/load_balancing_policies/client_side_weighted_round_robin/client_side_weighted_round_robin_lb_test.cc

```diff
@@ -0,0 +1,2047 @@
+      makeTestHost(info_, "tcp://127.0.0.1:82", simTime()),
+  };
+  simTime().setMonotonicTime(MonotonicTime(std::chrono::seconds(30)));
+  setHostClientSideWeight(hosts[0], 40, 5, 10);
+  setHostClientSideWeight(hosts[1], 41, 5, 10);
+  setHostClientSideWeight(hosts[2], 42, 5, 10);
+  // Setting client side weights should not change the host weights.
+  EXPECT_EQ(hosts[0]->weight(), 1);
+  EXPECT_EQ(hosts[1]->weight(), 1);
+  EXPECT_EQ(hosts[2]->weight(), 1);
+  // Update weights on hosts.
+  lb_->updateWeightsOnHosts(hosts);
+  // All hosts have client side weights, so the weights should be updated.
+  EXPECT_EQ(hosts[0]->weight(), 40);
+  EXPECT_EQ(hosts[1]->weight(), 41);
+  EXPECT_EQ(hosts[2]->weight(), 42);
+}
+
+TEST_P(ClientSideWeightedRoundRobinLoadBalancerTest, UpdateWeightsOneHostHasClientSideWeight) {
+  init(false);
+  HostVector hosts = {
+      makeTestHost(info_, "tcp://127.0.0.1:80", simTime()),
+      makeTestHost(info_, "tcp://127.0.0.1:81", simTime()),
+      makeTestHost(info_, "tcp://127.0.0.1:82", simTime()),
+  };
+  simTime().setMonotonicTime(MonotonicTime(std::chrono::seconds(30)));
+  // Set client side weight for one host.
+  setHostClientSideWeight(hosts[0], 42, 5, 10);
+  // Setting client side weights should not change the host weights.
+  EXPECT_EQ(hosts[0]->weight(), 1);
+  EXPECT_EQ(hosts[1]->weight(), 1);
+  EXPECT_EQ(hosts[2]->weight(), 1);
+  // Update weights on hosts.
+  lb_->updateWeightsOnHosts(hosts);
+  // Only one host has client side weight, other hosts get the median weight.
+  EXPECT_EQ(hosts[0]->weight(), 42);
+  EXPECT_EQ(hosts[1]->weight(), 42);
+  EXPECT_EQ(hosts[2]->weight(), 42);
+}
+
+TEST_P(ClientSideWeightedRoundRobinLoadBalancerTest, UpdateWeightsDefaultIsMedianWeight) {
+  init(false);
+  HostVector hosts = {
+      makeTestHost(info_, "tcp://127.0.0.1:80", simTime()),
+      makeTestHost(info_, "tcp://127.0.0.1:81", simTime()),
+      makeTestHost(info_, "tcp://127.0.0.1:82", simTime()),
+      makeTestHost(info_, "tcp://127.0.0.1:83", simTime()),
+      makeTestHost(info_, "tcp://127.0.0.1:84", simTime()),
+  };
+  simTime().setMonotonicTime(MonotonicTime(std::chrono::seconds(30)));
+  // Set client side weight for first three hosts.
+  setHostClientSideWeight(hosts[0], 5, 5, 10);
+  setHostClientSideWeight(hosts[1], 42, 5, 10);
+  setHostClientSideWeight(hosts[2], 5000, 5, 10);
+  // Setting client side weights should not change the host weights.
+  EXPECT_EQ(hosts[0]->weight(), 1);
+  EXPECT_EQ(hosts[1]->weight(), 1);
+  EXPECT_EQ(hosts[2]->weight(), 1);
+  EXPECT_EQ(hosts[3]->weight(), 1);
+  EXPECT_EQ(hosts[4]->weight(), 1);
+  // Update weights on hosts.
+  lb_->updateWeightsOnHosts(hosts);
+  // First three hosts have client side weight, other hosts get the median
+  // weight.
+  EXPECT_EQ(hosts[0]->weight(), 5);
+  EXPECT_EQ(hosts[1]->weight(), 42);
+  EXPECT_EQ(hosts[2]->weight(), 5000);
+  EXPECT_EQ(hosts[3]->weight(), 42);
+  EXPECT_EQ(hosts[4]->weight(), 42);
```

Comment:
the median weight is `42` here?

Reply:
Yes. `5` and `5000` are outliers, and `42` is the median (if we sort 5, 42, 42, 42, 5000, then 42 is in the middle.)